app/models/commit_checks.rb

```diff
@@ -0,0 +1,92 @@
+class CommitChecks
+  OUTPUT_TTL = 10.minutes.to_i
+  FINAL_STATUSES = %w(failed error success)
+
+  def initialize(commit)
+    @commit = commit
+  end
+
+  def run
+    self.status = 'running'
+    commands = StackCommands.new(stack)
+    commands.with_temporary_working_directory(commit: commit) do |directory|
+      deploy_spec = DeploySpec::FileSystem.new(directory, stack.environment)
+      Bundler.with_clean_env do
+        capture_all(build_commands(deploy_spec.dependencies_steps, chdir: directory))
+        capture_all(build_commands(deploy_spec.review_checks, chdir: directory))
+      end
+    end
+  rescue Command::Error
+    self.status = 'failed'
```

Comment:
Yeah the status is so the client side knows it's over. The command status etc is appended to the output.